### ISSUE TITLE
Implementation with widgets

### ISSUE THREAD

**Issue by arubinst** [jaguar-orm/sqflite#14]
TL;DR:
Could anybody provide a sample application that has different ListViews/forms etc.? I can run a simple example with just a main() but I would need inspiration for how to use this with different widgets and how to deal with all the async operations. Thanks!

Long version:
I'm having a really hard time figuring out how to use this out of a simple dart code with a single main.

I have a fairly complex DB structure with an app that should deal with several input forms and ListViews. I declared a singleton DBProvider class supposed to keep a connection open and to give me access to the adapter from any other class.

However, I really don't know how to deal with the async operations for my different beans. I guess what I really don't know is where I should put what...

Does anybody have a sample project that I could use as inspiration?

Thanks !

**Comment by arubinst** [comment]
Never mind. I think I kind of got it... I'm still lost with some issues and I don't know whether the solutions I found are the way to go, but I moved past the blocking I had.

**Comment by tejainece** [comment]
Hi,

What issues do you have?
